### PR TITLE
Cache context base base in quote driver

### PR DESCRIPTION
**Pull request by nicolasstucki**
This enables caching the compiler used of call `Expr.run`.

### CODE REVIEW

File: compiler/src/dotty/tools/dotc/quoted/QuoteDriver.scala [hunk 1]
```diff
@@ -2,20 +2,21 @@ package dotty.tools.dotc.quoted
...
 import java.net.URLClassLoader
 
-import Toolbox.{Run, Settings, Show}
 import dotty.tools.dotc.tastyreflect.TastyImpl
 
 class QuoteDriver extends Driver {
   import tpd._
 
-  def run[T](expr: Expr[T], settings: Settings[Run]): T = {
+  private[this] val contextBase: ContextBase = new ContextBase
```
Comment (smarter) [inline compiler/src/dotty/tools/dotc/quoted/QuoteDriver.scala]: I don't think this is thread-safe

Comment (nicolasstucki) [reply compiler/src/dotty/tools/dotc/quoted/QuoteDriver.scala]: It is not and not intended to be. Each thread should generate its own toolbox. I will add documentation and thick how to check that they do not use the same compiler in two threads.

Comment (nicolasstucki) [reply compiler/src/dotty/tools/dotc/quoted/QuoteDriver.scala]: Decided to make it thread-safe by synchronization instead. If someone wants parallelism they should create several instances of the `Toolbox`

**Comment by nicolasstucki** [comment]
Rebased

**Comment by nicolasstucki** [comment]
Rebased and reenabled test

**Comment by biboudis** [comment]
Also had a look. Seems good.

File: compiler/src/dotty/tools/dotc/quoted/QuoteDriver.scala (hunk 1 above)
Comment (smarter) [reply compiler/src/dotty/tools/dotc/quoted/QuoteDriver.scala]: > Decided to make it thread-safe by synchronization instead.

I don't see any synchronization logic in the code.

Comment (nicolasstucki) [reply compiler/src/dotty/tools/dotc/quoted/QuoteDriver.scala]: I must have lost that commit in one of the reading. I will add it again.

Comment (Blaisorblade) [reply compiler/src/dotty/tools/dotc/quoted/QuoteDriver.scala]: Searching for `synchronized` still finds nothing here?

Comment (nicolasstucki) [reply compiler/src/dotty/tools/dotc/quoted/QuoteDriver.scala]: It is in #4746

**Pull request merged**